### PR TITLE
xDai deployment

### LINKED ISSUE #111: Support xDAI
- Deploy contracts to xDAI
- Add contract addresses from xDAI to the `networks.ts` config file

### PR DESCRIPTION
**Pull request by cag**
Closes #111

### CODE REVIEW

File: src/config/networks.ts [hunk 1]
```diff
@@ -37,5 +37,12 @@ export const defaultNetworks: NetworksConfig = {
     proxyFactoryAddress: '0xfC7577774887aAE7bAcdf0Fc8ce041DA0b3200f7',
     multiSendAddress: '0x8D29bE29923b68abfDD21e541b9374737B49cdAD',
     fallbackHandlerAddress: '0x40A930851BD2e590Bd5A5C981b436de25742E980'
+  },
+  // xdai
+  100: {
+    masterCopyAddress: '0x6851D6fDFAfD08c0295C392436245E5bc78B0185',
```
Comment (rmeissner) [inline src/config/networks.ts:43]: Why do we not use the official mastercopy on xdai?

File: src/config/networks.ts [hunk 2]
```diff
@@ -37,5 +37,12 @@ export const defaultNetworks: NetworksConfig = {
...
+    proxyFactoryAddress: '0xfC7577774887aAE7bAcdf0Fc8ce041DA0b3200f7',
+    multiSendAddress: '0x8D29bE29923b68abfDD21e541b9374737B49cdAD',
+    fallbackHandlerAddress: '0xd5D82B6aDDc9027B22dCA772Aa68D5d74cdBdF44'
```
Comment (rmeissner) [inline src/config/networks.ts:46]: Why do we not use the official fallback handler on xdai?

File: truffle-config.js [hunk 3]
```diff
@@ -10,13 +10,15 @@ const networks = Object.assign(...[
   [4, 'rinkeby'],
   [5, 'goerli', `${2e9}`],
   [42, 'kovan'],
-].map(([networkId, network, gasPrice]) => ({
+  [77, 'sokol',, true],
```
Comment (rmeissner) [inline truffle-config.js:13]: Note: we do not have sokol in the networks file

**Review by rmeissner — CHANGES_REQUESTED** [review]

File: src/config/networks.ts (hunk 1 above)
Comment (cag) [reply src/config/networks.ts:43]: Wait, those are not the official addresses? I got them from https://github.com/gnosis/safe-contracts/blob/development/networks.json

Comment (rmeissner) [reply src/config/networks.ts:43]: Ahhh you use the v1.2.0 address ... then I would update this everywhere, not just for xdai. You should check by tag

Comment (germartinez) [reply src/config/networks.ts:43]: It's fine. The rest of the networks are updated here: https://github.com/gnosis/contract-proxy-kit/pull/126

File: src/config/networks.ts (hunk 2 above)
Comment (germartinez) [reply src/config/networks.ts:46]: Is not this the right one? https://github.com/gnosis/safe-contracts/blob/v1.2.0/networks.json#L118
From where did we get the fallback handler at `0x40A930851BD2e590Bd5A5C981b436de25742E980` for the other networks?

File: truffle-config.js (hunk 3 above)
Comment (germartinez) [reply truffle-config.js:13]: Do we want to create a "Sokol deployment PR" or remove this?

File: src/config/networks.ts (hunk 2 above)
Comment (rmeissner) [reply src/config/networks.ts:46]: so this seems to be the 1.1.1 fallback handler and the other is 1.1.0 (as mentioned on etherscan). Note 1.1.1 is the current. Code wise 1.1.0 and 1.1.1 should be the same.

See https://etherscan.io/address/0x40A930851BD2e590Bd5A5C981b436de25742E980
and https://etherscan.io/address/0xd5D82B6aDDc9027B22dCA772Aa68D5d74cdBdF44

**Review by germartinez — APPROVED** [review]

**Review by rmeissner — APPROVED** [review]

**Pull request merged**